R/AbstractGraphReporter.R

```diff
@@ -442,49 +448,20 @@ AbstractGraphReporter <- R6::R6Class(
             log_info('Resetting cached graph_viz...')
             private$cache$graph_viz <- NULL
             return(invisible(NULL))
-        },
-
-        # Identify orphan nodes
-        identify_orphan_nodes = function() {
-            orphan_nodes <- base::setdiff(
-                self$nodes[, node]
-                , unique(c(self$edges[, SOURCE], self$edges[, TARGET]))
-            )
-            
-            # If there are none, then will be character(0)
-            return(orphan_nodes)
-        },
-
-        graph_layout_functions = list(
-            "tree" = function(pkg_graph) {igraph::layout_as_tree(pkg_graph)},
-            "circle" = function(pkg_graph) {igraph::layout_in_circle(pkg_graph)}
-        ),
-
-        calculate_graph_layout = function(plotDT) {
-
-            log_info(paste("Calculating graph layout for type:", self$layout_type))
-
-            # Calculate positions for specified layout_type
-            graph_func <- private$graph_layout_functions[[self$layout_type]]
-            plotMat <- graph_func(self$pkg_graph)
-
-            # It might be important to get the nodes from pkg_graph so that they
-            # are in the same order as in plotMat?
-            coordsDT <- data.table::data.table(
-                node = names(igraph::V(self$pkg_graph))
-                , level = plotMat[, 2]
-                , horizontal = plotMat[, 1]
-            )
-
-            # Merge coordinates with plotDT
-            plotDT <- merge(
-                x = plotDT
-                , y = coordsDT
-                , by = 'node'
-                , all.x = TRUE
-            )
-
-            return(plotDT)
         }
+
     )
 )
+
+#' @title Available Graph Layout Functions from igraph
+#' @name igraphAvailableLayouts
+#' @description Returns available \link[igraph:layout_]{\code{igraph} layout function}
+#' names. These names can be passed to
+#' \code{\link[visNetwork:visIgraphLayout]{visNetwork::visIgraphLayout}} or set
+#' as the \code{layout_type} for any reporters that inherit from
+#' \code{\link{AbstractGraphReporter}}.
+#' @return a character vector of igraph layout function names
+#' @export
```

Comment:
I feel weird about exporting this as part of the package.

Since it's only one line of code, I'd rather we just document somewhere "run this to find valid values"

Reply:
@jameslamb I don't know if I agree. I think it's useful to anybody trying to interactively re-layout their visualization to have this available. 

@bburns632 thoughts?

Reply:
I don't disagree that it's useful, I feel very strongly that this does not belong in our package's namespace.

It's a one-liner we could document in package docs or a readme.